plugin/trino-open-policy-agent/src/test/java/io/trino/plugin/openpolicyagent/OpaAccessControlFilteringUnitTest.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openpolicyagent;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.google.common.collect.Maps;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemSecurityContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static io.trino.plugin.openpolicyagent.RequestTestUtilities.assertStringRequestsEqual;
+import static io.trino.plugin.openpolicyagent.TestHelpers.NO_ACCESS_RESPONSE;
+import static io.trino.plugin.openpolicyagent.TestHelpers.OK_RESPONSE;
+import static io.trino.plugin.openpolicyagent.TestHelpers.systemSecurityContextFromIdentity;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
```

Comment:
BTW, we are moving the codebase to asserj's assertThat

Reply:
Should I do that as part of this PR or would you prefer to do that in a follow up one?

Reply:
You can do it as a followup.  Just wanted to give you a heads up